src/core_plugins/kibana/server/lib/supported_features/get_supported_features.js

```diff
@@ -0,0 +1,25 @@
+import { checkFieldStatsApi } from './feature_checks';
+
+
+const featureChecks = [
+  checkFieldStatsApi,
+];
+
+export async function getSupportedFeatures(callWithRequest) {
+  const { nodes: nodesInfo } = await callWithRequest('nodes.info', {
+    filterPath: [
+      'nodes.*.name',
+      'nodes.*.version',
+    ],
+  });
+
+  const clusterInfo = {
+    nodes: nodesInfo,
+  };
+
+  const featureFlags = featureChecks.reduce((featureFlags, featureCheck) => (
```

Comment:
I think using this as a way for one feature check to depend on another is not the best way, since you would still require them to be in the correct order in the array (which the actual check file has no influence on. So imho I would build a (proper) dependency mechanism if it's needed :-)

Since the checks anyway don't collect the data, but only operate on the data passed in from this file, is there really a need for one check to set multiple flags. I can hardly think of an example where that check couldn't be split up into two different checks.

I would at the moment (without knowing what other checks might be there) favor code readability - which was also the actual reason I commented on this, since it was the part in the code I actually had to scroll two times up and down between files, to get it :-)

But again, it's just a suggestion and I am fine with whatever way you decide :-)